### DISCUSSION TITLE
is it possible to stop a encoding process remotly?

### DISCUSSION THREAD

**Discussion by efraimjer** [bluenviron/mediamtx#2898] [Questions]
### Question

Im converting a few ip cameras from rtsp/rtmp to hls and recording the streams using a node js server, but i need from time to time do stop the encoding process and then restart it later, by command, is there any option through the api to do that?

**Comment by aler9** [comment]
Hello, the server by itself doesn't perform any encoding, eventually it performs remuxing (changes protocol without touching the codec).

If you want to stop the remuxing process from RTSP/RTMP to HLS, you can use the API to start and stop the HLS server by using the `/config/global/patch` endpoint.

If on the other hand you are performing re-encoding by using `ffmpeg` or `gstreamer` inside `runOnReady` or any other hook, you can use the API to fill `runOnReady` with the command you are using to re-encode in order to start the encoding process, and use the API to clear `runOnReady` when you don't want to re-encode anymore.

[a comment by an automated account is not shown]
